src/app/state/importaccounts/saga.ts

```diff
@@ -71,33 +77,58 @@ function* enumerateAccountsFromMnemonic(action: PayloadAction<string>) {
   }
 }
 
-function* enumerateAccountsFromLedger() {
+function* ensureBalancesArePresent(action: PayloadAction<number>) {
+  const existingAccounts = yield* select(selectImportAccountsDisplayList)
+  yield* setStep(ImportAccountsStep.LoadingBalances)
+  for (let i = 0; i < existingAccounts.length; i++) {
+    const account = existingAccounts[i]
+    if (!account.balance) {
+      const balance = yield* call(getBalance, hex2uint(account.publicKey))
+      yield* put(
+        importAccountsActions.updateAccount({
+          index: account.index,
```

Comment:
do we really need account.index ? Data structure looks like 
0: { index: 0 ...}
1: { index: 1 ...} 
in UI you can extract index from derivation path if you need it in selection modal

Reply:
.... right. Simplified that.